src/lib/Hydra/Plugin/SlackNotification.pm

```diff
@@ -0,0 +1,93 @@
+package Hydra::Plugin::SlackNotification;
+
+use strict;
+use parent 'Hydra::Plugin';
+use HTTP::Request;
+use LWP::UserAgent;
+use Hydra::Helper::CatalystUtils;
+use JSON;
+
+sub buildFinished {
+    my ($self, $build, $dependents) = @_;
+    my $cfg = $self->{config}->{slack};
+    my @config = defined $cfg ? ref $cfg eq "ARRAY" ? @$cfg : ($cfg) : ();
+
+    my $baseurl = $self->{config}->{'base_uri'} || "http://localhost:3000";
+
+    # Figure out to which channelss to send notification.  For each channel
+    # we send one aggregate message.
+    my %channels;
+    foreach my $b ($build, @{$dependents}) {
+        my $prevBuild = getPreviousBuild($b);
+        my $jobName = showJobName $b;
+
+        foreach my $channel (@config) {
+            my $force = $channel->{force};
+            next unless $jobName =~ /^$channel->{jobs}$/;
+
+            # If build is cancelled or aborted, do not send email.
+            next if ! $force && ($b->buildstatus == 4 || $b->buildstatus == 3);
+
+            # If there is a previous (that is not cancelled or aborted) build
+            # with same buildstatus, do not send email.
+            next if ! $force && defined $prevBuild && ($b->buildstatus == $prevBuild->buildstatus);
+
+            $channels{$channel->{url}} //= { channel => $channel, builds => [] };
+            push @{$channels{$channel->{url}}->{builds}}, $b;
+        }
+    }
+
+    return if scalar keys %channels == 0;
+
+    my ($authors, $nrCommits) = getResponsibleAuthors($build, $self->{plugins});
+
+    # Send a message to each room.
+    foreach my $url (keys %channels) {
+        my $channel = $channels{$url};
+        my @deps = grep { $_->id != $build->id } @{$channel->{builds}};
+
+        my $imgBase = "http://hydra.nixos.org";
+        my $img =
+            $build->buildstatus == 0 ? "$imgBase/static/images/checkmark_16.png" :
```

Comment:
These images looks very bad since they are scaled up a lot in the slack interface, especially on high-DPI screens. You should probably go for the 256 variants instead.